python/cudf/cudf/core/dataframe.py

```diff
@@ -6482,11 +6519,15 @@ def select_dtypes(self, include=None, exclude=None):
             inclusion = set()
         # remove all exclude types
         inclusion = inclusion - exclude_subtypes
-
-        for k, col in self._data.items():
-            infered_type = cudf_dtype_from_pydata_dtype(col.dtype)
-            if infered_type in inclusion:
-                df._insert(len(df._data), k, col)
+        if inclusion:
+            for k, col in self._data.items():
+                infered_type = cudf_dtype_from_pydata_dtype(col.dtype)
+                if infered_type in inclusion:
+                    df._insert(len(df._data), k, col)
+        else:
+            df._data.rangeindex = self._data.rangeindex
+            df._data.multiindex = self._data.multiindex
+            df._data.label_dtype = self._data.label_dtype
```

Comment:
**question** Why not transfer this information for the `inclusion` case? Is it because that destroys the information?